server/src/test/java/io/crate/lucene/CommonQueryBuilderTest.java

```diff
@@ -224,8 +224,8 @@ public void testIdQuery() throws Exception {
         assertThat(query.toString(), is("_id:[ff 69 31]"));
 
         query = convert("_id = 1");
-        assertThat(query, instanceOf(TermQuery.class));
-        assertThat(query.toString(), is("_id:[fe 1f]"));
+        assertThat(query, instanceOf(GenericFunctionQuery.class));
+        assertThat(query.toString(), is("(_cast(_id, 'integer') = 1)"));
     }
```

Comment:
As `str<->num` cast swap is removed, one would need to explicitly use quotes, 
like in line 222 (`Query query = convert("_id = 'i1'")`) to make use of optimal plan.

However, if string column doesn't mean "it's actually a number", like with `_id` cast swap optimization wouldn't hurt. 

I see 2 options:
1.  Keep this fix and warn users in docs that they might want to use explicit types in where parameters.
2. Add a `force` parameter to `cast` function (also accommodate docs force param docs with example where it's needed - when string's physical meaning is a number and we don't want it to be compared alphabetically.

Reply:
I actually think, that 2 and 3 are not good options -
2. why one would need a `force` in `strCol::numeric > 3`?  swapping cast is internal detail, user should not be bothered with that, and force looks like weird here without that knowledge.
3. We still need to fix it, as it's wrong behaviour, even if fix leads to expensive query.